sql/planbuilder/parse_test.go

```diff
@@ -27,227 +28,269 @@ func TestPlanBuilder(t *testing.T) {
 		{
 			Query: "with cte(y,x) as (select x,y from xy) select * from cte",
 			ExpectedPlan: `
-SubqueryAlias
- ├─ name: cte
- ├─ outerVisibility: false
- ├─ cacheable: false
+Project
+ ├─ columns: [cte.y:1!null, cte.x:2!null]
  └─ Project
-     ├─ columns: [xy.x:1!null, xy.y:2!null]
-     └─ Table
-         ├─ name: xy
-         └─ columns: [x y z]
+     ├─ columns: [cte.y:1!null, cte.x:2!null]
+     └─ SubqueryAlias
+         ├─ name: cte
+         ├─ outerVisibility: false
+         ├─ cacheable: false
+         └─ Project
+             ├─ columns: [xy.x:1!null, xy.y:2!null]
+             └─ Project
+                 ├─ columns: [xy.x:1!null, xy.y:2!null, xy.z:3!null]
+                 └─ Table
+                     ├─ name: xy
+                     └─ columns: [x y z]
 `,
 		},
 		{
 			Query: "select * from xy where x = 2",
 			ExpectedPlan: `
 Project
  ├─ columns: [xy.x:1!null, xy.y:2!null, xy.z:3!null]
- └─ Filter
-     ├─ Eq
-     │   ├─ xy.x:1!null
-     │   └─ 2 (tinyint)
-     └─ Table
-         ├─ name: xy
-         └─ columns: [x y z]
+ └─ Project
```

Comment:
I tried a couple deadends, this only works if we pull projection pruning forward. Most direct way of doing that is after we start moving rules onto a new format. If someone hits a perf issue because of that I can do sooner.